lib/index.d.ts

```diff
@@ -1,4 +1,9 @@
 import type { ShowdownExtension } from "showdown";
 
-declare function showdownHighlight({ pre: Boolean }): ShowdownExtension[];
+declare type ShowdownHighlightOptions = {
+  pre: boolean
+  auto_detection: boolean
+}
+
+declare function showdownHighlight(options?: Partial<ShowdownHighlightOptions>): ShowdownExtension[];
```

Comment:
Since its implementation has default parameter and default destructured value, it should be optional.